lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -515,8 +512,13 @@ def fail_job(self, job_state):
         """
 
         # First we rescue the pods logs
-        with open(job_state.output_file, 'r') as outfile:
-            stdout_content = outfile.read()
+        try:
+            with open(job_state.output_file, 'r') as outfile:
+                stdout_content = outfile.read()
+        except IOError as e:
+            stdout_content = "<Failed to recuperate log for job {} >".format(job_state.job_id)
+            log.error("Failed to get stdout for job %s", job_state.job_id)
+            log.exception(e)
```

Comment:
I'd consolidate these two to `log.exception("Failed to get stdout for job %s", job_state.job_id)`